samples/ChatSample.Cli/Program.cs

```diff
@@ -0,0 +1,592 @@
+
+    private static readonly Dictionary<string, TypeInfo> TypeMap = new(StringComparer.OrdinalIgnoreCase)
+    {
+        { "int", new(typeof(int), x => int.Parse(x, null)) },
+        { "string", new(typeof(string), x => x) },
+        { "bool", new(typeof(bool), x => bool.Parse(x)) },
+        { "double", new(typeof(double), x => double.Parse(x, null)) }
+    };
+
+    private static readonly Dictionary<string, TypeInfo[]> MethodDefineMap = new(StringComparer.OrdinalIgnoreCase);
+
+    private static readonly Dictionary<string, (TypeInfo[] Params, TypeInfo Item)> StreamMethodDefineMap = new(StringComparer.OrdinalIgnoreCase);
+
+    private static readonly Dictionary<(string ConnectionId, string StreamId), (TypeInfo TypeInfo, Channel<object> Channel)> StreamMap = [];
+
+    private static Task Main(string[] args)
+    {
+        if (args.Length == 0)
+        {
+            Console.WriteLine("Interactive mode, type 'help' for a list of available commands.");
+        }
+        return MainLoop(ReadLines(string.Join(" ", args)));
+    }
+
+    private static async IAsyncEnumerable<string> ReadLines(string? first)
+    {
+        if (!string.IsNullOrEmpty(first))
+        {
+            yield return first;
+        }
+        List<string> commandsInFile = [];
+        while (true)
+        {
+            var line = (await Console.In.ReadLineAsync())?.Trim();
+            if (line == null)
+            {
+                yield break;
+            }
+            if (line.Length == 0)
+            {
+                continue;
+            }
+            if (await Try(LoadRegex(), line, m => Load(m, commandsInFile)))
+            {
+                foreach (var command in commandsInFile)
+                {
+                    yield return command;
+                }
+                commandsInFile.Clear();
+            }
+            yield return line;
+        }
+    }
+
+    private static async Task Load(Match m, List<string> commandsInFile)
+    {
+        var fileName = m.Groups[1].Value;
+        if (!File.Exists(fileName))
+        {
+            Console.WriteLine($"File not found: {fileName}");
+            return;
+        }
+        using var reader = new StreamReader(fileName, new FileStreamOptions { Options = FileOptions.Asynchronous });
+        string? line;
+        while ((line = await reader.ReadLineAsync()) != null)
+        {
+            line = line.Trim();
+            if (line.Length == 0)
+            {
+                continue;
+            }
+            commandsInFile.Add(line);
+        }
+    }
+
+    private static async Task MainLoop(IAsyncEnumerable<string> commands)
+    {
+        await foreach (var line in commands)
+        {
+            if (line.Equals("exit", StringComparison.OrdinalIgnoreCase))
+            {
+                break;
+            }
+            if ("help".Equals(line, StringComparison.OrdinalIgnoreCase))
+            {
+                Console.WriteLine("Available commands:");
```

Comment:
do you want to use `System.CommandLine` which emulator is using?

Reply:
No. There is no interactive mode.